react/features/base/conference/middleware.any.ts

```diff
@@ -330,12 +330,44 @@ function _conferenceJoined({ dispatch, getState }: IStore, next: Function, actio
  * @private
  * @returns {Object} The value returned by {@code next(action)}.
  */
-function _connectionEstablished({ dispatch }: IStore, next: Function, action: AnyAction) {
+async function _connectionEstablished({ dispatch, getState }: IStore, next: Function, action: AnyAction) {
     const result = next(action);
 
     // FIXME: Workaround for the web version. Currently, the creation of the
     // conference is handled by /conference.js.
-    typeof APP === 'undefined' && dispatch(createConference());
+    if (typeof APP === 'undefined') {
+        dispatch(createConference());
+
+        return result;
+    }
+
+    // TODO keep this here till we move tracks and conference management from
+    // conference.js to react.
+    const state = getState();
+    let localTracks = getLocalTracks(state['features/base/tracks']);
+
+    // Do not signal audio/video tracks if the user joins muted.
+    for (const track of localTracks) {
+        // Always add the audio track on Safari because of a known issue where audio playout doesn't happen
+        // if the user joins audio and video muted.
+        if (track.muted
+            && !(browser.isWebKitBased() && track.jitsiTrack && track.jitsiTrack.getType() === MEDIA_TYPE.AUDIO)) {
+            try {
+                await dispatch(replaceLocalTrack(track.jitsiTrack, null));
+            } catch (error) {
+                logger.error(`Failed to replace local track (${track.jitsiTrack}) with null: ${error}`);
+            }
+        }
+    }
+
+    // Re-fetch the local tracks after muted tracks have been removed above.
+    // This is needed, because the tracks are effectively disposed by the replaceLocalTrack and should not be used
+    // anymore.
+    localTracks = getLocalTracks(getState()['features/base/tracks']);
+
+    const jitsiTracks = localTracks.map((t: any) => t.jitsiTrack);
+
+    APP.conference.startConference(jitsiTracks).catch(logger.error);
```

Comment:
I think this one will be a problematic. Here's the use case that worries me:
 - We don connect trough the conference.init method in the use case where we are not on the prejoin screen.
 - We have the track creation and connect in Promise.all and as part of the whole promise we set the tracks
 - This will be executed once the connection is established but in the same time I don't think we have any guarantee that the track creation will be done and the tracks would have been set. 

At least before we were calling startConference once the connect has finished but also the tracks have been created. Is the change of the behavior intentional? Are you sure there won't be any issues? In practice now we can start the conference without the tracks....

Reply:
I think if this happens we will do one more O/A to add the media after the conference have been set up. IMHO we need to fix this.

Reply:
Looking into this. I will look for a quick workaround we may later redo all these when moving the tracks handling out of conference.js.

Reply:
Also I think this would break mobile isn't it?

Reply:
Oh discard this one. We return early for mobile.